library/src/scala/quoted/QuoteContext.scala

```diff
@@ -3696,21 +3696,7 @@ trait QuoteContext { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
 
   }
 
-  /** Type of a QuoteContext provided by a splice within a quote that took this context.
-   *  It is only required if working with the reflection API.
-   *
-   *  Usually it is infered by the quotes an splices typing. But sometimes it is necessary
-   *  to explicitly state that a context is nested as in the following example:
-   *
-   *  ```scala
-   *  def run(using qctx: QuoteContext)(tree: qctx.reflect.Tree): Unit =
-   *    def nested()(using qctx.Nested): Expr[Int] = '{  ${ makeExpr(tree) } + 1  }
-   *    '{  ${ nested() } + 2 }
-   *  def makeExpr(using qctx: QuoteContext)(tree: qctx.reflect.Tree): Expr[Int] = ???
-   *  ```
-   */
-  type Nested = QuoteContext {
-    val reflect: self.reflect.type
-  }
+  /** Type of a QuoteContext provided by a splice within a quote that took this context. */
+  type Nested = QuoteContext
```

Comment:
If this is the case, it seems we can remove this concept?

Reply:
I considered it. But if we leave it we can add support for some of those features in a future release. I will look into it later.